pkg/agent/flannel/setup.go

```diff
@@ -139,6 +154,24 @@ func createFlannelConf(nodeConfig *config.Node) error {
 	}
 	confJSON = strings.ReplaceAll(confJSON, "%backend%", backendConf)
 
+	netMode, err := findNetMode(nodeConfig.AgentConfig.ClusterCIDRs)
+	if err != nil {
+		logrus.Fatalf("Error checking netMode")
+		return err
+	}
+
+	if netMode == (ipv4 + ipv6) {
+		confJSON = strings.ReplaceAll(confJSON, "%DUALSTACK%", "true")
+		for _, cidr := range nodeConfig.AgentConfig.ClusterCIDRs {
+			if utilsnet.IsIPv6(cidr.IP) {
+				// Assuming there is going to be only one ipv6 range
```

Comment:
Recent versions of Kubernetes allow you to specify multiple non-overlapping, non-contiguous ipv4/ipv6 ranges. Is it too much work to allow that? We should at least handle it properly.

Reply:
This is a good point Brad! I wanted to look at this but forgot to do it. I don't think flannel supports multiple ranges but I need to verify this. If it does not, I guess we should fail with an error describing this, right?

Reply:
Interesting, I'll have to dig up the changeling entry I was thinking of. It's possible that one of the cloud providers has added support for this already but the core controller-manager nodeipam does not yet.

Reply:
I added the link in the comment, so that we don't forget this

Reply:
Here's what I was thinking of - it appears that this is GKE only and hasn't hit core Kubernetes yet:
https://cloud.google.com/kubernetes-engine/docs/how-to/multi-pod-cidr